sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/kafka/AbstractKafkaPropertiesBeanPostProcessor.java

```diff
@@ -63,16 +66,13 @@ abstract class AbstractKafkaPropertiesBeanPostProcessor<T> implements BeanPostPr
         KAFKA_OAUTH_CONFIGS = Collections.unmodifiableMap(configs);
     }
 
-    private final AzureGlobalProperties azureGlobalProperties;
-
-    AbstractKafkaPropertiesBeanPostProcessor(AzureGlobalProperties azureGlobalProperties) {
-        this.azureGlobalProperties = azureGlobalProperties;
-    }
+    private AzureGlobalProperties azureGlobalProperties;
 
     @SuppressWarnings("unchecked")
     @Override
     public Object postProcessBeforeInitialization(Object bean, String beanName) throws BeansException {
         if (needsPostProcess(bean)) {
+            azureGlobalProperties = applicationContext.getBean(AzureGlobalProperties.class);
```

Comment:
do we need to handle the case when the bean is not present?

Reply:
the getBean will throw exception when not found the global properties.

Reply:
But the other two BPP won't throw any exceptions, so do we need to throw the exception here?

Reply:
Compared with the previous implementation, here the global properties bean should be available, can we assume it's always exist? if not, then exception should be thrown.